systemtest/src/test/java/io/strimzi/systemtest/upgrade/kraft/KRaftStrimziUpgradeST.java

```diff
@@ -68,167 +69,172 @@ void testUpgradeKafkaWithoutVersion() throws IOException {
-        logPodImages(TestConstants.CO_NAMESPACE);
+        logClusterOperatorPodImage(clusterOperatorNamespaceName);
+        logComponentsPodImages(componentsNamespaceName);
 
         // Check if UTO is used before changing the CO -> used for check for KafkaTopics
-        boolean wasUTOUsedBefore = StUtils.isUnidirectionalTopicOperatorUsed(TestConstants.CO_NAMESPACE, eoSelector);
+        boolean wasUTOUsedBefore = StUtils.isUnidirectionalTopicOperatorUsed(componentsNamespaceName, eoSelector);
 
-        changeClusterOperator(upgradeData, TestConstants.CO_NAMESPACE);
+        changeClusterOperator(clusterOperatorNamespaceName, componentsNamespaceName, upgradeData);
 
         if (TestKafkaVersion.supportedVersionsContainsVersion(upgradeData.getDefaultKafkaVersionPerStrimzi())) {
-            waitForKafkaClusterRollingUpdate();
+            waitForKafkaClusterRollingUpdate(componentsNamespaceName);
         }
 
-        logPodImages(TestConstants.CO_NAMESPACE);
+        logClusterOperatorPodImage(clusterOperatorNamespaceName);
+        logComponentsPodImages(componentsNamespaceName);
 
         // Upgrade kafka
-        changeKafkaAndMetadataVersion(upgradeData, true);
-
-        logPodImages(TestConstants.CO_NAMESPACE);
-
-        checkAllImages(upgradeData, TestConstants.CO_NAMESPACE);
+        changeKafkaAndMetadataVersion(componentsNamespaceName, upgradeData, true);
+        logComponentsPodImages(componentsNamespaceName);
+        checkAllComponentsImages(componentsNamespaceName, upgradeData);
 
         // Verify that Pods are stable
-        PodUtils.verifyThatRunningPodsAreStable(TestConstants.CO_NAMESPACE, clusterName);
+        PodUtils.verifyThatRunningPodsAreStable(componentsNamespaceName, clusterName);
 
         // Verify upgrade
-        verifyProcedure(upgradeData, testStorage.getContinuousProducerName(), testStorage.getContinuousConsumerName(), TestConstants.CO_NAMESPACE, wasUTOUsedBefore);
+        verifyProcedure(componentsNamespaceName, upgradeData, testStorage.getContinuousProducerName(), testStorage.getContinuousConsumerName(), wasUTOUsedBefore);
     }
 
     @BeforeEach
     void setupEnvironment() {
         NamespaceManager.getInstance().createNamespaceAndPrepare(CO_NAMESPACE);
+
+        if (!CO_NAMESPACE.equals(TEST_SUITE_NAMESPACE)) {
+            NamespaceManager.getInstance().createNamespaceAndPrepare(TEST_SUITE_NAMESPACE);
+        }
     }
 
-    protected void afterEachMayOverride(ExtensionContext extensionContext) {
-        cleanUpKafkaTopics();
-        ResourceManager.getInstance().deleteResources();
+    @AfterEach
+    void afterEach() {
+        cleanUpKafkaTopics(TEST_SUITE_NAMESPACE);
+        deleteInstalledYamls(CO_NAMESPACE, TEST_SUITE_NAMESPACE, coDir);
         NamespaceManager.getInstance().deleteNamespaceWithWait(CO_NAMESPACE);
+
+        if (!CO_NAMESPACE.equals(TEST_SUITE_NAMESPACE)) {
+            NamespaceManager.getInstance().deleteNamespaceWithWait(TEST_SUITE_NAMESPACE);
+        }
```

Comment:
I see this in many places...(i.e., same code). You should think of some auxiliary method in this case...but before you do so please tell me why it is needed (I think also Lukas pointed on it)